projects/slap_manipulation/src/slap_manipulation/dataloaders/robot_loader.py

```diff
@@ -0,0 +1,668 @@
+    """train on a dataset from RLBench"""
+
+    def __init__(
+        self,
+        dirname,
+        template="*.h5",
+        predict: str = "",
+        verbose=False,
+        num_pts=10000,
+        data_augmentation=True,
+        random_idx=False,
+        random_cmd=True,
+        first_keypoint_only=False,
+        keypoint_range: list = [0, 1, 2],
+        show_voxelized_input_and_reference=False,
+        show_input_and_reference=False,
+        show_cropped=False,
+        ori_dr_range=np.pi / 4,
+        cart_dr_range=1.0,
+        first_frame_only=False,
+        trial_list: list = [],
+        orientation_type="quaternion",
+        multi_step=False,
+        crop_radius=True,
+        ambiguous_radius=0.03,
+        crop_radius_chance=0.75,
+        crop_radius_shift=0.1,
+        crop_radius_range=[0.3, 1.0],
+        visualize=False,
+        visualize_reg_targets=False,
+        yaml_file="./assets/language_variations/v0.yml",
+        dr_factor=1,
+        *args,
+        **kwargs,
+    ):
+        if yaml_file is not None:
+            self.annotations = load_annotations_dict(yaml_file)
+        else:
+            self.annotations = None
+        self.random_idx = random_idx
+        self.random_cmd = random_cmd
+        self.first_keypoint_only = first_keypoint_only
+        self.keypoint_to_use = None
+        self.multi_step = multi_step
+        self.num_pts = num_pts
+        self.data_augmentation = data_augmentation
+        self.ori_dr_range = ori_dr_range
+        self.cart_dr_range = cart_dr_range
+        self.keypoint_range = keypoint_range
+        self.ori_type = orientation_type
+        self.trial_list = trial_list
+        self.crop_radius = crop_radius
+        self.crop_radius_shift = crop_radius_shift
+        self.crop_radius_range = crop_radius_range
+        self.crop_radius_chance = crop_radius_chance
+        self._cr_min = crop_radius_range[0]
+        self._cr_max = crop_radius_range[1]
+        self._cr_rng = self._cr_max - self._cr_min
+        self._ambiguous_radius = ambiguous_radius
+
+        # super(RoboPenDataset, self).__init__(
+        super(RLBenchDataset, self).__init__(
+            dirname,
+            template,
+            verbose,
+            trial_list=trial_list,
+            TrialType=RPHighLevelTrial,
+        )
+        self._voxel_size = 0.001
+        self._voxel_size_2 = 0.01
+        self._local_problem_size = 0.1
+        self.num_pts = num_pts
+        self.DEBUG = True
+        self.predict = predict  # can be "contact" or "release"
+
+        # configuration and data files
+        self.cam_mapping_json_path = "./assets/robopen08_mapping.json"
+        self.calibration_json_path = "./assets/robopen08_calibration.json"
+        self.intrinsics_json_path = "./assets/robopen08_intrinsics.json"
+        self.cam_intrinsics, self.cam_extrinsics = self.read_cam_config()
+
+        self.debug_closest_pt = False
+        self.robot = FrankaPanda()
```

Comment:
does this actually do anything

Reply:
Just for adding/reverting grasp offsets. Will edit this once I am working on Stretch data to reflect Stretch model parameters.